app/javascript/modCenter/__tests__/moderationArticles.test.jsx

```diff
@@ -0,0 +1,93 @@
+import { h } from 'preact';
+import { render, fireEvent } from '@testing-library/preact';
+import { ModerationArticles } from '../moderationArticles';
+
+const getTestArticles = () => {
+  const articles = [
+    {
+      id: 1,
+      title: 'An article title',
+      path: 'an-article-title-di3',
+      published_at: '2019-06-22T16:11:10.590Z',
+      cached_tag_list: 'discuss, javascript, beginners',
+      user: {
+        articles_count: 1,
+        name: 'hello',
+        id: 1,
+      },
+    },
+    {
+      id: 2,
+      title:
+        'An article title that is quite very actually rather extremely long with all things considered',
+      path:
+        'an-article-title-that-is-quite-very-actually-rather-extremely-long-with-all-things-considered-fi8',
+      published_at: '2019-06-24T09:32:10.590Z',
+      cached_tag_list: '',
+      user: {
+        articles_count: 3,
+        name: 'howdy',
+        id: 2,
+      },
+    },
+  ];
+  return JSON.stringify(articles);
+};
+
+describe('<ModerationArticles />', () => {
+  beforeEach(() => {
+    render(
+      <div
+        class="mod-index-list"
+        id="mod-index-list"
+        data-articles={getTestArticles()}
+      >
+        <div class="flag-user-modal-container hidden" />
+      </div>,
+    );
+    render(<ModerationArticles />, document.getElementById('mod-index-list'));
+  });
+
+  it('renders a list of 2 articles', () => {
+    const listOfArticles = document.querySelectorAll(
```

Comment:
Instead of using `document.querySelectorAll` opt for preact-testing-library `getByTestId` function. I have an example of it in the single article tests. below, https://github.com/thepracticaldev/dev.to/pull/8995/files#diff-befec6d46653590452227f93b5954784R146, e.g. `const button = getByTestId(`mod-article-${article.id}`);`

For the reasoning of using a test ID attribute, see [Making your UI tests resilient to change](https://kentcdodds.com/blog/making-your-ui-tests-resilient-to-change/)

The same can be done in other tests below.